src/xmipp/bindings/python/xmipp_base.py

```diff
@@ -98,7 +110,321 @@ def tryRun(self):
+
+    @staticmethod
+    def installEnvironCmd(environName, installCmdOptions=None, **kwargs):
+        """ expected kwargs:  see xmipp_conda_envs.py
+                pythonVersion: number, if None (default) no python is installed
+                dependencies: list, 'depName=ver' or just 'depName' ([] default)
+                channels: list, channel names where looking for
+                defaultInstallOptions: dict, to be converted in command formating
+                                             (e.g. gpuTag=-gpu, {} by default)
+                pipPackages: list, pipName==ver' or just 'pipName' ([] default)
+        """
+        # Preparing options from kwargs
+        pyVer = kwargs.get('pythonVersion', None)
+        python = "python="+pyVer if pyVer else ""
+
+        deps = ' '.join([dep for dep in kwargs.get('dependencies', [])])
+
+        chs = kwargs.get('channels', [])
+        chFlags = (" -c %s" % " -c ".join([c for c in chs]) if len(chs) > 0 else "")
+
+        options = installCmdOptions or kwargs.get('defaultInstallOptions', {})
+        pipPack = kwargs.get('pipPackages', [])
+
+        # Composing the commands
+        cmdList = []
+        cmdList.append("export PYTHONPATH=\"\"")  # TODO: consider if from kwargs
+        cmdList.append(CondaEnvManager.getCondaActivationCmd())
+        cmdList.append("conda create -q --force --yes -n %s %s %s %s"
+                       % (environName, python, deps, chFlags))
+        cmdList.append("conda activate %s" % environName)
+        if pipPack:
+            cmdList.append("pip install %s" % " ".join([dep for dep in pipPack]))
+        cmdList.append("conda env export > %s.yml" % environName)
+
+        cmd = ' && '.join(cmdList)
+        try:
+            cmd = cmd % options
+        except KeyError as ex:  # chr(37) = %
+            print("Option not found constructing the conda installing commnad:\n"
+                  "%s  %s  (%s)" % (cmd, chr(37), ', '.join(options.keys())))
+
+        return cmd, environName
+
+
+def getModel(*modelPath, **kwargs):
+    """ Returns the path to the models folder followed by
+        the given relative path.
+    .../xmipp/models/myModel/myFile.h5 <= getModel('myModel', 'myFile.h5')
+
+        NOTE: it raise and exception when model not found, set doRaise=False
+              in the arguments to skip that raise, especially in validation
+              asserions!
+    """
+    model = getXmippPath('models', *modelPath)
+
+    # Raising an error to prevent posterior errors and to print a hint
+    if kwargs.get('doRaise', True) and not os.path.exists(model):
+        raise Exception("'%s' model not found. Please, run: \n"
+                        " > scipion installb deepLearningToolkit" % modelPath[0])
+    return model
+
+
+# FIXME: XmippMdRow is almost the same than pwem.emlib.metadata.utils.Row
+# FIXME:  It's here because deep_denoising needs it and pwem can't be imported
+class XmippMdRow:
```

Comment:
Why not? I guess that if you have decided that pwem cannot be included, you are right, but it would be good to leave some comment about why.

Reply:
No, it's not my decision. It's not possible to import any Scipion's module inside conda environments, at least not in an elegant and general way.

The reason:
 
Since Scipion is in a virtual enviroment and we want to use it from another virtual enviramoent, the unique way is to add in the PYTHONPATH the scipion's site-packages. Then, finding it's site-package is not elegant but possible. However, if Scipion is installed in devel mode (`pip install -e /path/to/scipion-repos` what we do to be able to keep developing inside an installed pip package), that repos don't end in site-packages but egg-links do. These egg-links are recognized by the python where they was installed but not for any python. Thus, this results in a 'module not found' when trying to import any of them...

Reply:
Another option would be to install scipion-em inside the environment that will need pyworkflow and not sure if still, it would be usable. I think there is no easy way to do it and duplicating code might be the simplest. Maybe in future versions we can keep extracting things and have a scipion-em-md (package)...or maybe even better a xmipp-em-md that could be installed in both environments.